util/oidc/oidc.go

```diff
@@ -336,6 +346,41 @@ func (a *ClientApp) HandleLogin(w http.ResponseWriter, r *http.Request) {
 	http.Redirect(w, r, url, http.StatusSeeOther)
 }
 
+// getAzureKubernetesServiceAccountToken returns the specified file's content, which is expected to be a Kubernetes service account token.
+// Kubernetes is responsible for updating the file as service account tokens expire.
+func (a *ClientApp) getAzureKubernetesServiceAccountToken(context.Context) (string, error) {
```

Comment:
This method should use `rest.InClusterConfig()` instead of AZURE_FEDERATED_TOKEN_FILE to get the mounted service account. To be able to run tests, you should provide an Interface that can be mocked.

Reply:
We can use rest.InClusterCOnfig to call into API server, but I will still need to read the file to get the token
rest.InCLusterConfig will have to read the pod spec to same environment variable and then read the file. It can be done directly using the env variable

We can rename the method as it is confusing we are not getting the service account token but federated token from azure